tensorboard/pip_package/build_pip_package.sh

```diff
@@ -12,66 +12,92 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# ==============================================================================
-
-
 set -e
+if [ -z "${RUNFILES}" ]; then
+  RUNFILES="$(CDPATH= cd -- "$0.runfiles" && pwd)"
+fi
 
-die() {
-  printf >&2 '%s\n' "$1"
-  exit 1
+if [ "$(uname)" = "Darwin" ]; then
+  sedi="sed -i ''"
+else
+  sedi="sed -i"
+fi
+
+smoke() {
+  virtualenv -qp python$1 venv$1
+  cd venv$1
+  . bin/activate
+  pip install -qU pip
+  pip install -qU tf-nightly
+  pip install -qU ../dist/*py$1*.whl >/dev/null
+  mkdir logs
+  mkfifo pipe
+  tensorboard --port=0 --logdir=logs 2>pipe &
```

Comment:
Won't piping stderr to the fifo mean that we're swallowing any errors that TensorBoard logs if it does in fact fail to start up?  Or do those somehow get displayed by the perl command below reading the pipe?  (I'm not that familiar with perl and it's hard to google, so a comment would be helpful here.)

Reply:
Yes. `print STDERR` in the perl one-liner is sort of like `| tee /dev/stderr`. I could write an essay explaining the depths of this subroutine. It's not unusual from the perspective of these tools. I'd prefer it stay tiny, without comment.